File: app/src/test/java/br/com/santanderteste/presenter/LoginPresenterTest.java
package br.com.santanderteste.presenter;

import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;
import org.mockito.junit.MockitoJUnit;
import org.mockito.junit.MockitoRule;

import br.com.santanderteste.ui.interfaces.ILoginView;
import br.com.santanderteste.ui.interfaces.IUserRepository;

import static org.mockito.Mockito.times;
import static org.mockito.Mockito.verify;

public class LoginPresenterTest {

    @Rule
    public MockitoRule mockitoRule = MockitoJUnit.rule();

    private LoginPresenter loginPresenter;
    @Mock
    IUserRepository userRepository;
    @Mock
    ILoginView loginView;

    private String username;
    private String password;

    @Before
    public void setUp() {
        MockitoAnnotations.initMocks(this);

        loginPresenter = new LoginPresenter(loginView, userRepository);

    }

    @Test
    public void shouldCall_RequestUserData() {

        username = "51544156049";
        password = "Test@1";

        loginPresenter.checkLoginData(username, password);

        verify(loginPresenter, times(1)).requestUserData(username, password);
    }

    @Test
    public void loadUserData() {
    }

    @Test
    public void requestUserData() {
    }
}

File: app/src/test/java/com/earaujo/santander/login/LoginPresenterTest.kt
package com.earaujo.santander.login

import com.earaujo.santander.login.LoginActivityInput
import com.earaujo.santander.login.LoginPresenter
import com.earaujo.santander.repository.Resource
import com.earaujo.santander.repository.models.LoginResponse
import com.nhaarman.mockito_kotlin.*
import org.junit.Before
import org.junit.Test
import org.mockito.Mock
import org.mockito.MockitoAnnotations
import java.lang.ref.WeakReference

class LoginPresenterTest {

    @Mock
    lateinit var loginActivityInput: LoginActivityInput

    @Before
    fun setup() {
        MockitoAnnotations.initMocks(this)
    }

    @Test
    fun `when presentLoginResponse should call loginCallback with the same parameter received`() {
        //Prepare
        val loginPresenter = LoginPresenter()
        val loginResponse: Resource<LoginResponse> = Resource.success(null)
        loginPresenter.loginActivityInput = WeakReference(loginActivityInput)

        //Action
        loginPresenter.presentLoginResponse(loginResponse)

        //Verification
        verify(loginActivityInput, times(1)).loginCallback(loginResponse)
    }

    @Test
    fun `when presentSetUserName should call displayUserName with the same parameter received`() {
        //Prepare
        val loginPresenter = LoginPresenter()
        val userName = "username"
        loginPresenter.loginActivityInput = WeakReference(loginActivityInput)

        //Action
        loginPresenter.presentSetUserName(userName)

        //Verification
        verify(loginActivityInput, times(1)).displayUserName(userName)
    }
}